src/cmd/singularity/cli/actions.go

```diff
@@ -255,13 +255,21 @@ func execWrapper(cobraCmd *cobra.Command, image string, args []string, name stri
 		}
 	}
 
+	// Copy and cache environment
+	environment := os.Environ()
+
+	// Clean environment
 	if !IsCleanEnv {
-		for _, env := range os.Environ() {
+		for _, env := range environment {
 			e := strings.SplitN(env, "=", 2)
 			if len(e) != 2 {
 				sylog.Verbosef("can't process environment variable %s", env)
 				continue
 			}
+			// skip any variables prefixed with "SINGULARITYENV_"
+			if strings.HasPrefix(e[0], "SINGULARITYENV_") {
+				continue
```

Comment:
I don't think we want to continue when we find the `SINGULARITYENV_` prefixed variables. IIRC, we *only* want to pass through variables which are prefixed with `SINGULARITYENV_` (and also any special cases such as `HOME`).

Reply:
the skip is to avoid something like
```
[eduardo@rhel7 singularity]$ export SINGULARITYENV_ISSUE=1883
[eduardo@rhel7 singularity]$ ./builddir/singularity exec /tmp/alpine_3.7.sif env | grep ISSUE
SINGULARITYENV_ISSUE=1883
ISSUE=1883
```